docs/resampling.ipynb

```diff
@@ -0,0 +1,380 @@
+    "        bg_rms = np.array([sep.Background(band).globalrms for band in img])\n",
+    "    else:\n",
+    "        bg_rms =  sep.Background(detect).globalrms\n",
+    "    return catalog, bg_rms\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "obs_hdu = fits.open('../data/test_resampling/Cut_HSC.fits')\n",
+    "data_hsc = obs_hdu[0].data.byteswap().newbyteorder()\n",
+    "wcs_hsc = WCS(obs_hdu[0].header)\n",
+    "psf_hsc = fits.open('../data/test_resampling/PSF_HSC.fits')[0].data\n",
+    "\n",
+    "\n",
+    "\n",
+    "x,y = np.where(psf_hsc[0]*0==0)\n",
+    "Np1, Np2 = psf_hsc[0].shape\n",
+    "\n",
+    "hst_hdu = fits.open('../data/test_resampling/Cut_HST.fits')\n",
+    "data_hst = hst_hdu[0].data\n",
+    "wcs_hst = WCS(hst_hdu[0].header)\n",
+    "print(wcs_hst.wcs.crval)\n",
+    "wcs_hst.wcs.crval += 2.4750118475607095e-05*np.array([-np.cos(0.4136047623181346 ), -np.sin(0.4136047623181346 )])\n",
+    "print(wcs_hst.wcs.crval)\n",
+    "psf_hst = fits.open('../data/test_resampling/PSF_HST.fits')[0].data\n",
+    "np1,np2 = np.shape(psf_hst)\n",
+    "psf_hst = psf_hst.reshape(1,np1,np2)\n",
+    "\n",
+    "noise = np.concatenate((np.concatenate((data_hst[0,:], data_hst[:,0])),\n",
+    "                                            np.concatenate((data_hst[-1,:], data_hst[:,-1]))))\n",
+    "\n",
+    "n1,n2 = np.shape(data_hst)\n",
+    "data_hst = data_hst.reshape(1, n1, n2).byteswap().newbyteorder()*np.max(data_hsc)/np.max(data_hst)\n",
+    "\n",
+    "r, N1, N2 = data_hsc.shape\n",
+    "\n",
+    "catalog_hst, bg_rms_hst = makeCatalog(data_hst)\n",
+    "catalog_hsc, bg_rms_hsc = makeCatalog(data_hsc)\n",
+    "\n",
+    "#plotting setup\n",
+    "#norm = #scarlet.display.Asinh(img=data_hsc, Q=20)\n",
+    "# Map i,r,g -> RGB\n",
+    "filter_indices = [2,0,1]\n",
+    "\n",
+    "xo,yo = catalog_hst['x'], catalog_hst['y']\n",
+    "\n",
+    "ra, dec = wcs_hst.wcs_pix2world(yo,xo,0)\n",
+    "Yo,Xo, l = wcs_hsc.wcs_world2pix(ra, dec, 0, 0)\n",
+    "\n",
+    "img_rgb = scarlet.display.img_to_rgb(data_hsc)\n",
+    "plt.subplot(121)\n",
+    "plt.imshow(img_rgb)\n",
+    "plt.plot(Xo,Yo, 'o')\n",
+    "plt.subplot(122)\n",
+    "plt.imshow(np.arcsinh(data_hst[0]))\n",
+    "#plt.plot(xo,yo, 'o')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initialize the sources\n",
+    "Each source is a list of fundamental `scarlet.Component` instances and must be based on `scarlet.Source` or a derived class, in this case `ExtendedSource`, which enforces that the source is monotonic and symmetric."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "obs_hst = scarlet.Observation(data_hst, wcs = wcs_hst, psfs = None, structure = np.array([0,0,0,1]))\n",
```

Comment:
@fred3m Not to be addressed in this PR, but something we should correct: `psfs=None` is the only way to say: don't convolve the model. What we should rather say `if psfs is None or psfs = blend.psfs: don't convolve`

Then we could properly say that this Observation actually has a PSF, it just is the same one as our target.

Reply:
I don't see what you gain by that. Not specifying a PSF is just another way of saying we are not doing a convolution, so the result is exactly the same. I think it is simpler to say that if the PSF of the scene and the model are the same then initialize the `Observation` with `psfs=None`.

Reply:
We'd gain consistency. If we want to match PSFs (from model to observation), we give the PSF argument, if not, we omit it. So far so, good.

What happens if I want to change the PSF of `Blend`, i.e. of our internal model? I now need to remember to specify the PSFs for all observations. If I forget, the result will be incorrect. Therefore the observation should be fully specified. If an Observation now has the same PSF as `Blend`, it won't convolve. It's a very simple test that adds consistency to the declaration of observations:
```
obs_hst = scarlet.Observation(data_hst, wcs = wcs_hst, psfs = psfs_hst)
obs_hsc = scarlet.LowResObservation(data_hsc, wcs = wcs_hsc, psfs = psfs_hsc)
```

Reply:
I agree but I think that this only works if we create a `PSF` class. Otherwise it has to check the PSFs element by element, which is ~43^2 pixels for an HSC PSF and much larger when it is upsampled to HST resolution.

Reply:
I'm pretty sure it would just do a reference check because both point to the same PSF structure